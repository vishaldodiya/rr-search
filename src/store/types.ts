import {PinsState} from "./pins/types";
import {DataState} from "./data/types";
import {AnalyticsState} from "./analytics/types";

export interface StoreState {
    pins: PinsState,
    data: DataState,
    analytics: AnalyticsState
};