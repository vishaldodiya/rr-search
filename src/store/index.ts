import {createStore, combineReducers} from "redux";
import * as constants from "../constants";
import DataReducer from "./data/reducer";
import PinsReducer from "./pins/reducer";
import AnalyticsReducer from "./analytics/reducer";
import * as calendar from "../acme-search/calendar.json";
import * as contacts from "../acme-search/contacts.json";
import * as dropbox from "../acme-search/dropbox.json";
import * as slack from "../acme-search/slack.json";
import * as tweet from "../acme-search/tweet.json";

const data = {
    calendar: calendar.calendar,
    contacts: contacts.contacts,
    dropbox: dropbox.dropbox,
    slack: slack.slack,
    tweet: tweet.tweet
};

const reducer = combineReducers({
    pins: PinsReducer,
    data: DataReducer,
    analytics: AnalyticsReducer
});

const store = createStore(reducer);

store.dispatch({
    type: constants.LOAD_DATA,
    list: data
})

store.subscribe(() => console.log(store.getState()));

export default store;