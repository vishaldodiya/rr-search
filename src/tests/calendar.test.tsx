import * as React from "react";
import Calendar from "../containers/calendar";
import {Provider} from "react-redux";
import store from "../store/index";
import * as renderer from 'react-test-renderer';

 
describe('Calendar', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Calendar data={
                {
                    "id": "12345",
                    "title": "Acme Proposal Meeting",
                    "invitees": "dave, john, bob, carol",
                    "date": "2019-01-10 10:00:00",
                    "uid": "calendar-1234"
                }}
            ></Calendar>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});