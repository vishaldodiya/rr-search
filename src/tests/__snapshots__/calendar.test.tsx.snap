// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar snapshot renders 1`] = `
<article
  className="calendar results__card card"
  id="calendar-1234"
>
  <div
    className="card__actions"
  >
    <span>
      <i
        className="far fa-calendar-alt"
      />
    </span>
    <button
      className="card__pin"
      onClick={[Function]}
    >
      <i
        className="fas fa-thumbtack c--green"
      />
    </button>
  </div>
  <h4>
    Acme Proposal Meeting
  </h4>
  <div
    className="calendar__info"
  >
    <div
      className="calendar__info--date"
    >
      <span>
        on January 10, 2019 at 10:00 AM
      </span>
    </div>
    <div
      className="calendar__info--invitees"
    >
      <i
        className="fas fa-users c--gray"
      />
      <span
        className="card__list"
      >
        dave
      </span>
      <span
        className="card__list"
      >
        john
      </span>
      <span
        className="card__list"
      >
        bob
      </span>
      <span
        className="card__list"
      >
        carol
      </span>
    </div>
  </div>
</article>
`;
